modin/pandas/groupby.py

```diff
@@ -294,6 +296,34 @@ def first(self, **kwargs):
     def backfill(self, limit=None):
         return self.bfill(limit)
 
+    _internal_by_cache = no_default
+
+    # TODO: since python 3.9:
+    # @cached_property
```

Comment:
If this is needed in several places let's make a separate PR to introduce our own caching decorator.
I mean it's really easy, like (untested):
```python
if sys.version_info >= (3, 9):
    caching_property = functools.caching_property
else:
    def caching_property(func):
        cache_attr = f"__{func.__name__}_cache"
        @functools.wraps(func)
        def wrap(self, *args, **kw):
            try:
                return getattr(self, cache_attr)
            except AttributeError:
                pass
            value = func(self, *args, **kw)
            setattr(self, cache_attr, value)
            return value
        return wrap
```